boards/arm/nrf9160dk_nrf52840/doc/index.rst

```diff
@@ -123,37 +123,116 @@ Board controller firmware
+| P0.01, P0.00, P0.15, and P0.14 | VCOM2                            |
++--------------------------------+----------------------------------+
+| P0.02                          | LED1                             |
++--------------------------------+----------------------------------+
+| P0.03                          | LED2                             |
++--------------------------------+----------------------------------+
+| P0.04                          | LED3                             |
++--------------------------------+----------------------------------+
+| P0.05                          | LED4                             |
++--------------------------------+----------------------------------+
+| P0.08                          | Switch 1                         |
++--------------------------------+----------------------------------+
+| P0.09                          | Switch 2                         |
++--------------------------------+----------------------------------+
+| P0.06                          | Button 1                         |
++--------------------------------+----------------------------------+
+| P0.07                          | Button 2                         |
++--------------------------------+----------------------------------+
+| P0.17, P0.18, and P0.19        | Arduino pin headers              |
++--------------------------------+----------------------------------+
+| P0.21, P0.22, and P0.23        | Trace interface                  |
++--------------------------------+----------------------------------+
+| COEX0, COEX1, and COEX2        | COEX interface                   |
++--------------------------------+----------------------------------+
+
+For a complete list of all the routing options available,
+see the `nRF9160 DK board control section in the nRF9160 DK User Guide`_.
+
+If you want to route some of the above pins differently or enable any of the
+other available routing options, enable or disable the devicetree node that
+represents the analog switch that provides the giving routing.
+For example, if you want to enable the optional routing for the nRF9160 pins
+P0.17, P0.18, and P0.19 so that they are routed to nRF52840 pins P0.17, P0.20,
+and P0.15, respectively, add the following in the devicetree overlay in your
+application:
+
+.. code-block:: none
+
+   &nrf_interface_pins_0_2_routing {
+           status = "okay";
+   };
+
+And if you want to, for example, disable routing for the VCOM2 pins, add the
+following:
+
+.. code-block:: none
+
+   &vcom2_pins_routing {
+           status = "disabled";
+   };
+
+
+The following devicetree nodes are defined for the analog switches present
+on the nRF9160 DK:
+
++------------------------------------+------------------------------+
+| Devicetree node alias              | Analog switch name           |
++====================================+==============================+
+| ``vcom0_pins_routing``             | nRF91_UART1 (nRF91_APP1)     |
```

Comment:
Did you consider naming these to match the analog switch name more closely, e.g. "nrf91_uart1" instead of "vcom0_pins_routing", "nrf91_led1" instead of "led1_pin_routing", etc.? If so, why did you decide against it?

Reply:
I thought about it but I was afraid it could be confusing. Especially for the nRF91_UART1 switch which in fact provides routing for `uart0` in the nrf9160dk_nrf9160 board. I wanted to provide more meaningful names as node labels. The nodes themselves are named exactly after the names of analog switches from the [DK documentation](https://infocenter.nordicsemi.com/topic/ug_nrf91_dk/UG/nrf91_DK/board_controller.html). The above node is named `switch-nrf91-uart1` and in the generated `zephyr.dts` file the user will see:
```
		vcom0_pins_routing: switch-nrf91-uart1 {
			compatible = "nordic,nrf9160dk-optional-routing";
			control-gpios = < &gpio1 0xe 0x1 >;
			status = "okay";
		};
```
I think it should be clear enough.
But I'm open to change the naming of those nodes labels if the current one is actually confusing or inconvenient.